src/tools/cephfs/top/cephfs-top

```diff
@@ -201,29 +207,45 @@ class FSTop(object):
         self.topl.addstr(0, 0, ITEMS_PAD.join(heading), curses.A_STANDOUT | curses.A_BOLD)
         return x_coord_map
 
+    def has_metrics(self, metadata, metrics_keys):
+        for key in metrics_keys:
+            if key not in metadata:
+                return False
+        return True
+
     def refresh_client(self, client_id, metrics, counters, client_meta, x_coord_map, y_coord):
         for item in MAIN_WINDOW_TOP_LINE_ITEMS_END:
             coord = x_coord_map[item]
             if item == FS_TOP_MAIN_WINDOW_COL_MNTPT_HOST_ADDR:
-                self.mainw.addstr(y_coord, coord[0],
-                                  f'{client_meta[CLIENT_METADATA_MOUNT_POINT_KEY]}@'
-                                  f'{client_meta[CLIENT_METADATA_HOSTNAME_KEY]}/'
-                                  f'{client_meta[CLIENT_METADATA_IP_KEY]}')
+                if self.has_metrics(client_meta, [CLIENT_METADATA_MOUNT_POINT_KEY,
+                                                  CLIENT_METADATA_HOSTNAME_KEY,
+                                                  CLIENT_METADATA_IP_KEY]):
+                    self.mainw.addstr(y_coord, coord[0],
+                                      f'{client_meta[CLIENT_METADATA_MOUNT_POINT_KEY]}@'
+                                      f'{client_meta[CLIENT_METADATA_HOSTNAME_KEY]}/'
+                                      f'{client_meta[CLIENT_METADATA_IP_KEY]}')
         for item in MAIN_WINDOW_TOP_LINE_ITEMS_START:
+            if not self.has_metrics(client_meta, [CLIENT_METADATA_VALID_METRICS_KEY]):
+                # eliminate invalid CLIENT_ID display
+                continue
             coord = x_coord_map[item]
             hlen = coord[1] - len(ITEMS_PAD)
             if item == FS_TOP_MAIN_WINDOW_COL_CLIENT_ID:
                 self.mainw.addstr(y_coord, coord[0],
                                   wrap(client_id.split('.')[1], hlen))
-            elif item == FS_TOP_MAIN_WINDOW_COL_MNT_ROOT:
+            elif (item == FS_TOP_MAIN_WINDOW_COL_MNT_ROOT and
+                  self.has_metrics(client_meta, [CLIENT_METADATA_MOUNT_ROOT_KEY])):
                 self.mainw.addstr(y_coord, coord[0],
                                   wrap(client_meta[CLIENT_METADATA_MOUNT_ROOT_KEY], hlen))
         cidx = 0
         for item in counters:
+            if not self.has_metrics(client_meta, [CLIENT_METADATA_VALID_METRICS_KEY]):
+                cidx += 1
+                continue
             coord = x_coord_map[item]
             m = metrics[cidx]
             typ = MAIN_WINDOW_TOP_LINE_METRICS[MGR_STATS_COUNTERS[cidx]]
-            if item.lower() in client_meta['valid_metrics']:
+            if self.has_metrics(client_meta[CLIENT_METADATA_VALID_METRICS_KEY], [item.lower()]):
```

Comment:
I don't think this check is required since counters would always be present for a client

Reply:
I have seen `valid_metrics` missing for a client and I've fixed it in the latest code. You mean only check if the `valid_metrics` is present and no need to check if  the counters are there?

Reply:
Dropped altogether, as you suggested

Reply:
> I have seen `valid_metrics` missing for a client and I've fixed it in the latest code. You mean only check if the `valid_metrics` is present and no need to check if the counters are there?

Right. It's the counters check that can be removed (for now).